packages/flutter_tools/lib/src/runner/flutter_command.dart

```diff
@@ -374,6 +375,13 @@ abstract class FlutterCommand extends Command<void> {
             "Normally there's only one, but when adding Flutter to a pre-existing app it's possible to create multiple.");
   }
 
+  void usesDeviceUserOption() {
+    argParser.addOption(FlutterOptions.kDeviceUser,
+      help: 'Target user identifier on Android.\n'
```

Comment:
I went through about 4 different versions of this help text, but none of it worked for every command I added it to.  It's too sparse though, I'll change this to make the help text different per command.

You might want to use this because you have multiple users on the device (System > Multiple Users) and only want to install to one user OR because you have [work profiles](https://developer.android.com/work/managed-profiles#testing_tips).

> For example, to find the users on a device, you would run this command:
```
$ adb shell pm list users
UserInfo{0:Drew:13} running
UserInfo{10:Work profile:30} running
```
> In this case, the primary user("Drew") has the user ID 0, and the work profile has the user ID 10. To run an app in the work profile, you would use a command like this:
```
$ adb shell am start --user 10 \
-n "com.example.myapp/com.example.myapp.testactivity" \
-a android.intent.action.MAIN -c android.intent.category.LAUNCHER
```

Reply:
don't hesitate to add a big, huge help text that encompasses every possible command

Reply:
In fact, rather than give each command a different flavor I would consider just adding headings to the flag:

```
with run:

...

with install:

...

```

Reply:
So you run `flutter attach -h` and see the details for `flutter run` and `flutter drive` etc?

Reply:
well, I would hide this particular option under verboseHelp, so -h v. 

Maybe adding them all together is a bad idea, but I'd be worried about keeping the documentation up to date. Maybe a shared header + command specific details added on

Reply:
How about:
`'Identifier number for a user or work profile on Android only. Run "adb shell pm list users" for available identifiers.'`

What are our criteria for hiding options?  This one may be uncommonly used, but it's not like experimental or only relevant for Flutter or g3 development (like local engine flags, etc).

Reply:
usually just go with your gut